File: WordPress/Classes/Services/LocalCoreDataService.m
#import "LocalCoreDataService.h"

@interface LocalCoreDataService ()
@property (nonatomic, strong, readwrite) NSManagedObjectContext *managedObjectContext;
@end

@implementation LocalCoreDataService

- (instancetype)initWithManagedObjectContext:(NSManagedObjectContext *)context
{
    NSParameterAssert([context isKindOfClass:[NSManagedObjectContext class]]);
    
    self = [super init];
    if (self) {
        _managedObjectContext = context;
    }
    
    return self;
}

@end

File: Sources/WordPressData/Swift/LocalCoreDataService.swift
import CoreData

@objc open class LocalCoreDataService: NSObject {
    @objc public let managedObjectContext: NSManagedObjectContext

    @objc public init(managedObjectContext context: NSManagedObjectContext) {
        self.managedObjectContext = context
        super.init()
    }
}
